packages/@aws-cdk/aws-appmesh/lib/virtual-node-listener.ts

```diff
@@ -216,4 +244,23 @@ class VirtualNodeListenerImpl extends VirtualNodeListener {
       },
     });
   }
+
+  private renderTls(tls: IListenerTlsConfig): CfnVirtualNode.ListenerTlsProperty {
+
+    if (this.isAcmCertificate(tls.certificate)) {
+      tls.certificate = new AcmTlsCertificate({
+        acmCertificate: tls.certificate.attrArn,
+      });
+    }
+
+    return {
+      certificate: tls.certificate.bind(),
+      mode: tls.mode,
+    };
+  }
+
+  private isAcmCertificate(cert: TlsCertificate | CfnCertificate): cert is CfnCertificate {
```

Comment:
Refer to this comment by Adam, https://github.com/aws/aws-cdk/pull/10879#discussion_r505096464
CDK team has a design tenet in the L2 layer that they don't use Typescript's Union type.

Reply:
hm interesting awslint doesn't catch this https://github.com/aws/aws-cdk/blob/master/DESIGN_GUIDELINES.md#unions

Reply:
This might be exempt because this is not a construct API.

> *Do not use TypeScript* *union types in construct APIs* (string | number) since many of the target languages supported by the CDK cannot strongly-model such types. Instead use static methods.

However I think there is a more CDK-esk way of doing this.